src/stores/roles/actions.ts

```diff
@@ -0,0 +1,27 @@
+import { doRequest, get, post, put } from '@/services/http'
+import { getRoles } from './getters'
+import type { Role } from '@/domain/Role'
+import { roles } from './state'
+
+function setRoles(newRoles: Role[]): void {
+  roles.value = newRoles
+}
+
+async function fetchRoles(): Promise<void> {
+  if (getRoles?.value?.length > 0) {
+    return
+  }
+
+  const fetchedRoles = await get<Role[]>('/roles')
+
+  setRoles(fetchedRoles)
+}
+
+async function quizResult(quizAnswer: any) {
```

Comment:
```suggestion
async function quizResult(quiz: Quiz) {
```

Use the type you have created inside domain folder